[TACHYON-1428] Added checkstyle rule for documentation of public methods and classes

HorizonNet opened the pull request:
https://tachyon.atlassian.net/browse/TACHYON-1428

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Tachyon-Pull-Request-Builder/8166/
Test PASSed.

jsimsa commented on build/checkstyle/suppressions.xml:
```diff
@@ -0,0 +1,6 @@
+<suppressions>
+    <suppress files="[\\/]src[\\/]test[\\/].*" id="MethodNameRegular" />
+    <suppress files="[\\/]src[\\/]main[\\/].*" id="MethodNameTest" />
+</suppressions>
```
add a missing new line

apc999 replied:
also, could you explain each suppress rule in comment, so we know what is suppressed for what?

Thanks

apc999 commented on build/checkstyle/tachyon_checks.xml:
```diff
@@ -223,6 +223,14 @@
     </module>
     <module name="MissingDeprecated"/>
     <module name="EqualsHashCode"/>
+    <module name="JavadocMethod">
```
@HorizonNet , could you add some comment here to explain this rule and the rule below ?
We should keep the checks config file well documented, so we will try to comment other modules too.

apc999 commented:
@HorizonNet thanks for the awesome work and actually all the effort before this PR to fix/add javadoc in this project in order to enable this stylecheck. It turns out really exciting for me. Thanks so much!

I and @jsimsa left a few comments, mostly asking for some comments in the config. After that, we will merge the PR. Thanks!

HorizonNet commented:
@apc999 Thanks for the comments. I updated the PR with some comments about the suppressions and the rules. I also changed the name of the suppressions to be more meaningful.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/Tachyon-Pull-Request-Builder/8172/
Test PASSed.

apc999 commented:
@HorizonNet thanks! This is awesome!
+1 LGTM

jsimsa commented:
LGTM
